ambari-common/src/main/python/resource_management/libraries/functions/mpack_manager_helper.py

```diff
@@ -31,19 +31,16 @@
 MPACK_VERSION_KEY_NAME = 'mpack_version'
 MODULE_VERSION_KEY_NAME = 'module_version'
 
-
 def get_component_conf_path(mpack_name, instance_name, module_name, components_instance_type,
                             subgroup_name='default', component_instance_name='default'):
   """
   :returns the single string that contains the path to the configuration folder of given component instance
   :raises ValueError if the parameters doesn't match the mpack or instances structure
   """
 
-  conf_json = get_conf_dir(mpack_name, instance_name, subgroup_name, module_name,
+  return get_conf_dir(mpack_name, instance_name, subgroup_name, module_name,
                            {components_instance_type: [component_instance_name]})
 
-  return conf_json[COMPONENTS_PLURAL_KEY_NAME][components_instance_type.lower()][COMPONENT_INSTANCES_PLURAL_KEY_NAME][
```

Comment:
This code has totally broken, there are two issues here:
1. No way to access conf_json dict directly via "components" key, you have to use "mpacks/mpack_name/...", but they are not always passed in, these arguments can be None
2. The original code return a string only for single mpack instance, but we actually need consider mutiple mpack instances, so the new code return a list